src/Transports.AspNetCore/Common/GraphQLRequestDeserializerBase.cs

```diff
@@ -0,0 +1,36 @@
+using GraphQL.Server.Common;
+using Microsoft.AspNetCore.Http;
+using System.IO;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace GraphQL.Server.Transports.AspNetCore.Common
+{
+    /// <summary>
+    /// Abstract base deserializer of GraphQL requests, implementing any shared logic.
+    /// </summary>
+    /// <remarks>
+    /// Various forms of requests should be supported by a GraphQL HTTP endpoint.
+    /// See https://graphql.org/learn/serving-over-http/ for specifics.
+    /// </remarks>
+    public abstract class GraphQLRequestDeserializerBase : IGraphQLRequestDeserializer
+    {
+        public abstract Task<GraphQLRequestDeserializationResult> DeserializeFromJsonBodyAsync(HttpRequest httpRequest, CancellationToken cancellationToken = default);
+
+        public abstract GraphQLRequest DeserializeFromQueryString(IQueryCollection queryCollection);
+
+        public async Task<GraphQLRequest> DeserializeFromGraphBodyAsync(Stream bodyStream)
```

Comment:
Why use shared base class with a few lines of code when it can be done in middleware?

Reply:
I do not see the point in the existence of this class, it only creates unnecessary restrictions.

Reply:
Yea I wasn't really ready for you to review :P I was going back and fwd between needing serialization and not, but it turns out for application/graphql the content is just the raw query value, nothing else. So I've put it back in the middleware.

Reply:
By the way, only 43/55 files checked now. Review, review never changes...